src/components/rightCol/ThisWeek.vue

```diff
@@ -1,30 +1,46 @@
 <template>
-  <h2 class="custom-card-sub-title">Transactions</h2>
-  <TransactionSubCard v-for="transaction in getTrs"
-                      :key="transaction" :amount="transaction.amount" 
-                      :symbol="transaction.currency"
-                      :desc="transaction.description"
-                      :date="dateFormated(transaction.date)"
-                      :unformatedDate="transaction.date"
-                      :name="transaction.relatedUser ? transaction.relatedUser.display : transaction.related.type.name "
-                      picto="QR"/>
+  <div class="transactions-loader">
+    <loading  v-model:active="getTransactionsBatchLoading"
+              :can-cancel="false"
+              :is-full-page="false"/>
+  </div>
+  <div v-if="!getTransactionsBatchLoading">
+    <div v-if="getTrs?.length === 0" class="custom-card-title mb-3" > Aucune transaction dans votre historique</div>
+    <div v-else >
+      <h2  class="custom-card-sub-title">Transactions</h2>
+      <TransactionSubCard v-for="transaction in getTrs"
+                          :key="transaction" :amount="transaction.amount"
+                          :symbol="transaction.currency"
+                          :desc="transaction.description"
+                          :date="dateFormated(transaction.date)"
+                          :unformatedDate="transaction.date"
+                          :name="transaction.relatedUser ? transaction.relatedUser.display : transaction.related.type.name "
+                          picto="QR"/>
+    </div>
+  </div>
 </template>
 
 <script lang="ts">
   import { Options, Vue } from 'vue-class-component';
   import TransactionSubCard from "./TransactionSubCard.vue"
+  import Loading from 'vue-loading-overlay';
+  import 'vue-loading-overlay/dist/vue-loading.css';
 
   @Options({
     name:"ThisWeek",
     components: {
-      TransactionSubCard
+      TransactionSubCard,
+      Loading: Loading
     },
     data() {
       return {}
     },
     computed: {
       getTrs(): any {
         return this.$store.state.lokapi.thisWeektransactions
+      },
+      getTransactionsBatchLoading(): boolean {
+        return this.$store.state.lokapi.transactionsBatchLoading
```

Comment:
Computed properties are not getters per se. I prefer if you stick to the convention of naming them like a property, not a method: ie `isLoadingTransactionsBatch` for instance.

Reply:
Don't get me wrong, I would also like to change the `getTrs` computed prop above, which is not right to me also, and I can assume that you named this computed prop like you did because of it. It is legacy code from the original authors of the app.

Reply:
You are right, i'm working on the needed changes ...

Reply:
@SeddikKadi thanks. That looks good to me. 👍